src/app/pages/swap/components/swap-review/swap-review.tsx

```diff
@@ -0,0 +1,17 @@
+import { useParams } from 'react-router-dom';
+
+import { BitcoinSwapReview } from './bitcoin-swap-review';
+import { StacksSwapReview } from './stacks-swap-review';
+
+export function SwapReview() {
+  const { origin } = useParams();
+
+  switch (origin) {
+    case 'bitcoin':
+      return <BitcoinSwapReview />;
+    case 'stacks':
+      return <StacksSwapReview />;
+    default:
+      return null;
+  }
+}
```

Comment:
Shouldn't this be defined in the routes directly?

Reply:
Good point, I'll move it.

Reply:
If you follow the suggestion I made on not using dynamic params, this would ofc have to change to read origin from context.

But, switch statements like this I would try to avoid. They add another logic fork deep in the feature component tree. If these use different components, couldn't this also be passed in at the `generateSwapRoutes` level, to avoid the switch?